jerry-core/rcs/rcs-records.cpp

```diff
@@ -0,0 +1,636 @@
+
+  const uint8_t *data_space_begin_p = rcs_get_node_data_space (rec_set_p, node_p);
+  const size_t data_space_size = rcs_get_node_data_space_size ();
+
+  const uint8_t *record_start_p = (const uint8_t *) rec_p;
+  const size_t record_size = rcs_record_get_size (rec_p);
+
+  const size_t record_offset_in_node = (size_t) (record_start_p - data_space_begin_p);
+  const size_t node_size_left = data_space_size - record_offset_in_node;
+
+  if (node_size_left > record_size)
+  {
+    return (rcs_record_t *) (record_start_p + record_size);
+  }
+
+  node_p = rec_set_p->get_next (node_p);
+  JERRY_ASSERT (node_p != NULL || record_size == node_size_left);
+
+  size_t record_size_left = record_size - node_size_left;
+  while (record_size_left >= data_space_size)
+  {
+    JERRY_ASSERT (node_p != NULL);
+
+    node_p = rec_set_p->get_next (node_p);
+    record_size_left -= data_space_size;
+  }
+
+  if (node_p == NULL)
+  {
+    JERRY_ASSERT (record_size_left == 0);
+    return NULL;
+  }
+
+  return (rcs_record_t *) (rcs_get_node_data_space (rec_set_p, node_p) + record_size_left);
+} /* rcs_record_get_next */
+
+/**
+ * Compares two charset records for equality.
+ *
+ * @return  true if strings inside records are equal
+ *          false otherwise
+ */
+bool
+rcs_record_is_equal (rcs_record_set_t *rec_set_p, /**< recordset */
+                     rcs_record_t *l_rec_p, /**< left record */
+                     rcs_record_t *r_rec_p) /**< rigth record */
+{
+  size_t l_rec_length = rcs_record_get_length (l_rec_p);
+  size_t r_rec_length = rcs_record_get_length (r_rec_p);
+
+  if (l_rec_length != r_rec_length)
+  {
+    return false;
+  }
+
+  rcs_iterator_t l_rec_it_ctx = rcs_iterator_create (rec_set_p, l_rec_p);
+  rcs_iterator_t r_rec_it_ctx = rcs_iterator_create (rec_set_p, r_rec_p);
+
+  rcs_iterator_skip (&l_rec_it_ctx, RCS_CHARSET_HEADER_SIZE);
+  rcs_iterator_skip (&r_rec_it_ctx, RCS_CHARSET_HEADER_SIZE);
+
+  lit_utf8_size_t i;
+  for (i = 0; i < l_rec_length; ++i)
+  {
+    lit_utf8_byte_t l_chr;
+    lit_utf8_byte_t r_chr;
+
+    rcs_iterator_read (&l_rec_it_ctx, &l_chr, sizeof (lit_utf8_byte_t));
+    rcs_iterator_read (&r_rec_it_ctx, &r_chr, sizeof (lit_utf8_byte_t));
+
+    if (l_chr != r_chr)
+    {
+      return false;
+    }
+
+    rcs_iterator_skip (&l_rec_it_ctx, sizeof (lit_utf8_byte_t));
+    rcs_iterator_skip (&r_rec_it_ctx, sizeof (lit_utf8_byte_t));
+  }
+
+  return true;
+} /* rcs_record_is_equal */
+
+/**
+ * Compare a record with a string (which could contain '\0' characters) for equality.
+ *
+ * @return  true if compared instances are equal
+ *          false otherwise
+ */
+bool
+rcs_record_is_equal_utf8 (rcs_record_set_t *rec_set_p, /**< recordset */
+                          rcs_record_t *rec_p, /**< record */
+                          const lit_utf8_byte_t *str_p, /**< string to compare with */
+                          lit_utf8_size_t str_size) /**< length of the string */
```

Comment:
As the function compares charset in a record and charset in a buffer, without any UTF-8 specifics, its better to rename the function to something like `rcs_record_is_equal_charset`.